#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.0044736246467309195 0.9491005964838286 0.3149413349126481 3.1685542628817998
  position -0.4481901895627004 1.3114189873715543 -1.0982410741861242
  follow "e-puck"
}
Background {
  backUrl [
    "textures/cubic/mountains_back.png"
  ]
  bottomUrl [
    "textures/cubic/mountains_bottom.png"
  ]
  frontUrl [
    "textures/cubic/mountains_front.png"
  ]
  leftUrl [
    "textures/cubic/mountains_left.png"
  ]
  rightUrl [
    "textures/cubic/mountains_right.png"
  ]
  topUrl [
    "textures/cubic/mountains_top.png"
  ]
  backIrradianceUrl [
    "textures/cubic/mountains_back.hdr"
  ]
  bottomIrradianceUrl [
    "textures/cubic/mountains_bottom.hdr"
  ]
  frontIrradianceUrl [
    "textures/cubic/mountains_front.hdr"
  ]
  leftIrradianceUrl [
    "textures/cubic/mountains_left.hdr"
  ]
  rightIrradianceUrl [
    "textures/cubic/mountains_right.hdr"
  ]
  topIrradianceUrl [
    "textures/cubic/mountains_top.hdr"
  ]
  luminosity 0
}
TexturedBackgroundLight {
}
DEF Track Solid {
  translation 0.240001 0.0125 1.3
  rotation 0.5773509358560258 -0.577349935856137 -0.577349935856137 2.09439
  scale 0.001 0.001 0.001
  children [
    DEF SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
        name "color_cccccc"
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            12.4323 -2182.57 -21.3934, 12.4323 -2195.97 -42.4676, 12.4323 -2222.38 -18.7471, 12.4323 -2208.08 2.05836, 12.4323 -2248.97 10.4641, 12.4323 -2156.7 -40.1423, 12.4323 -2168.93 -61.7594, 12.4323 -1516.24 669.094, 12.4323 -1545.9 653.918, 12.4323 -1559.71 685.085, 12.4323 -2234.09 31.4494, 12.4323 -1503.23 636.559, 12.4323 -2276.72 46.849, 12.4323 -2131.65 -54.037, 12.4323 -2112.39 -88.8333, 12.4323 -2142.61 -76.2922, 12.4323 -850.096 89.345, 12.4323 -906.753 83.1384, 12.4323 -908.492 107.948, 12.4323 -2261.7 68.6908, 12.4323 -2307.12 93.2028, 12.4323 -2102.61 -65.9804, 12.4323 -2076.52 -99.5355, 12.4323 -2292.44 116.87, 12.4323 -2335.71 143.276, 12.4323 -1478.23 652.142, 12.4323 -2067.52 -76.1967, 12.4323 -1467.86 618.483, 12.4323 -2030.78 -108.691, 12.4323 -850.968 114.156, 12.4323 -1444.66 633.479, 12.4323 -1439.85 600.326, 12.4323 -2319.35 165.636, 12.4323 -2358.77 190.759, 12.4323 -2021.6 -85.0139, 12.4323 -1984.31 -114.779, 12.4323 -1321.19 530.517, 12.4323 -1323.53 500.596, 12.4323 -1358.98 530.519, 12.4323 -2340.76 211.605, 12.4323 -2376.63 235.393, 12.4323 -1414.58 612.379, 12.4323 -1410.26 576.788, 12.4323 -1972.12 -91.1084, 12.4323 -1382.64 585.591, 12.4323 -1289.17 506.301, 12.4323 -1292.67 478.911, 12.4323 -2357 254.554, 12.4323 -2389.69 277.162, 12.4323 -1259.31 487.989, 12.4323 -1261.65 461.162, 12.4323 -1911.88 -120.986, 12.4323 -2368.5 294.492, 12.4323 -2398.36 316.177, 12.4323 -792.042 91.7188, 12.4323 -1902.01 -96.7295, 12.4323 -1228.93 473.23, 12.4323 -1229.9 446.814, 12.4323 -2375.7 331.541, 12.4323 -2403.04 352.579, 12.4323 -1850.82 -123.318, 12.4323 -1842.87 -98.5535, 12.4323 -1197.44 461.615, 12.4323 -1196.93 435.548, 12.4323 -2379.26 369.82, 12.4323 -792.02 116.527, 12.4323 -2404.35 392.354, 12.4323 -1164.31 452.926, 12.4323 -1158.45 426.473, 12.4323 -1798.17 -122.184, 12.4323 -2378.94 408.804, 12.4323 -2401.8 432.974, 12.4323 -1792.14 -96.9562, 12.4323 -1752.53 -117.926, 12.4323 -1129.39 447.154, 12.4323 -1122.2 421.449, 12.4323 -2374.72 448.465, 12.4323 -1748.52 -92.2977, 12.4323 -2395.32 473.915, 12.4323 -730.751 114.935, 12.4323 -731.622 90.1364, 12.4323 -1713.15 -110.926, 12.4323 -1094.46 444.667, 12.4323 -1086.45 419.774, 12.4323 -2366.58 488.152, 12.4323 -1708.82 -84.4317, 12.4323 -1689.62 -105.081, 12.4323 -2384.96 514.467, 12.4323 -1061.79 445.588, 12.4323 -1053.14 421.484, 12.4323 -1671.7 -73.9726, 12.4323 -1663.74 -97.4742, 12.4323 -672.892 109.917, 12.4323 -674.452 85.1259, 12.4323 -2354.64 527.008, 12.4323 -2370.9 553.734, 12.4323 -2339.21 563.985, 12.4323 -1033.02 449.628, 12.4323 -2353.53 590.682, 12.4323 -1023.59 426.284, 12.4323 -1575.98 -67.8147, 12.4323 -2322.78 594.692, 12.4323 -604.269 100.071, 12.4323 -606.208 75.2379, 12.4323 -1582.04 -43.6917, 12.4323 -1008.78 456.243, 12.4323 -2335.48 621.031, 12.4323 -995.684 434.581, 12.4323 -2304.86 621.346, 12.4323 -1524.53 -52.2078, 12.4323 -2316.26 646.892, 12.4323 -982.787 467.127, 12.4323 -1530.15 -28.0273, 12.4323 -968.111 446.723, 12.4323 -518.732 83.6756, 12.4323 -519.815 58.5542, 12.4323 -2285.87 643.75, 12.4323 -2295.82 668.887, 12.4323 -1471.69 -39.7484, 12.4323 -952.45 483.873, 12.4323 -936.176 464.723, 12.4323 -2265.68 662.501, 12.4323 -1476.28 -15.3669, 12.4323 -2274.06 687.474, 12.4323 -355.008 46.8467, 12.4323 -359.191 22.3567, 12.4323 -881.144 527.819, 12.4323 -2244.12 678.036, 12.4323 -2250.83 702.976, 12.4323 -861.925 510.323, 12.4323 -275.391 29.7995, 12.4323 -279.451 5.31716, 12.4323 -208.256 18.2702, 12.4323 -211.801 -6.28189, 12.4323 -823.428 559.964, 12.4323 -2220.96 690.657, 12.4323 -2225.98 715.608, 12.4323 -1392.39 -0.250995, 12.4323 -1385.91 -24.3328, 12.4323 -801.404 543.47, 12.4323 -151.232 -13.4083, 12.4323 -148.401 11.2457, 12.4323 -95.8348 -16.5068, 12.4323 -93.8702 8.26364, 12.4323 -760.898 590.906, 12.4323 -2195.9 700.531, 12.4323 -2199.26 725.485, 12.4323 -739.984 573.183, 12.4323 -47.9551 -16.0194, 12.4323 -46.8332 8.8654, 12.4323 0 -12.0687, 12.4323 -2168.6 707.702, 12.4323 -2167.1 733.25, 12.4323 -702.82 615.823, 12.4323 -683.923 596.534, 12.4323 -2134.77 712.666, 12.4323 0 12.9313, 12.4323 -651.754 634.035, 12.4323 -1303.17 -11.2074, 12.4323 -635.142 613.203, 12.4323 -2130.99 737.918, 12.4323 -606.696 646.514, 12.4323 -591.963 624.411, 12.4323 -2093.16 714.903, 12.4323 -1274.46 18.792, 12.4323 -565.148 654.548, 12.4323 -552.054 631.331, 12.4323 -2087.26 739.778, 12.4323 -1231.88 1.68174, 12.4323 -525.704 658.784, 12.4323 -514.073 634.547, 12.4323 -487.441 659.553, 12.4323 -477.081 634.35, 12.4323 -2041.36 714.005, 12.4323 -2033.31 738.457, 12.4323 -1211.34 31.622, 12.4323 -445.759 656.774, 12.4323 -434.065 630.462, 12.4323 -395.145 649.885, 12.4323 -379.35 622.093, 12.4323 -1973.32 709.542, 12.4323 -1172.55 15.738, 12.4323 -1159.87 44.9249, 12.4323 -1939.76 731.777, 12.4323 -305.457 634.697, 12.4323 -312.456 610.669, 12.4323 -1895.05 703.981, 12.4323 -280.768 631.426, 12.4323 -272.062 605.63, 12.4323 -1094.35 38.5437, 12.4323 -257.515 629.404, 12.4323 -229.182 603.554, 12.4323 -1880.67 728.18, 12.4323 -214.825 628.359, 12.4323 -182.547 604.464, 12.4323 -1830.5 702.323, 12.4323 -1086.09 66.9839, 12.4323 -149.905 630.67, 12.4323 -130.161 606.157, 12.4323 -1815.79 727.093, 12.4323 -1774.43 702.309, 12.4323 -114.513 630.511, 12.4323 -95.5222 604.175, 12.4323 -1025.81 58.4064, 12.4323 -78.4918 627.129, 12.4323 -45.8086 597.074, 12.4323 -1028.56 83.3366, 12.4323 -16.7837 618.305, 12.4323 -7.39334 592.444, 12.4323 15.4669 615.798, 12.4323 12.3755 590.99, 12.4323 -1744.05 726.492, 12.4323 -1726.71 700.584, 12.4323 -964.378 72.989, 12.4323 -1706.68 723.405, 12.4323 -1688.65 695.515, 12.4323 -966.741 97.8302, 12.4323 -1667.06 716.48, 12.4323 -1647.74 686.245, 12.4323 -1622.56 704.959, 12.4323 -1595.4 670.667
          ]
        }
        coordIndex [
          0, 1, 2, -1, 3, 0, 2, -1, 3, 2, 4, -1, 5, 1, 0, -1, 5, 6, 1, -1, 7, 8, 9, -1, 10, 3, 4, -1, 7, 11, 8, -1, 10, 4, 12, -1, 13, 6, 5, -1, 13, 14, 15, -1, 13, 15, 6, -1, 16, 17, 18, -1, 19, 10, 12, -1, 19, 12, 20, -1, 21, 22, 14, -1, 21, 14, 13, -1, 23, 20, 24, -1, 23, 19, 20, -1, 25, 11, 7, -1, 26, 22, 21, -1, 25, 27, 11, -1, 26, 28, 22, -1, 29, 16, 18, -1, 30, 31, 27, -1, 32, 24, 33, -1, 32, 23, 24, -1, 30, 27, 25, -1, 34, 28, 26, -1, 35, 28, 34, -1, 36, 37, 38, -1, 39, 33, 40, -1, 39, 32, 33, -1, 41, 31, 30, -1, 41, 42, 31, -1, 43, 35, 34, -1, 44, 38, 42, -1, 44, 36, 38, -1, 44, 42, 41, -1, 45, 46, 37, -1, 47, 40, 48, -1, 45, 37, 36, -1, 47, 39, 40, -1, 49, 50, 46, -1, 51, 35, 43, -1, 49, 46, 45, -1, 52, 48, 53, -1, 52, 47, 48, -1, 54, 16, 29, -1, 55, 51, 43, -1, 56, 57, 50, -1, 56, 50, 49, -1, 58, 53, 59, -1, 58, 52, 53, -1, 60, 51, 55, -1, 61, 60, 55, -1, 62, 63, 57, -1, 62, 57, 56, -1, 64, 58, 59, -1, 65, 54, 29, -1, 66, 64, 59, -1, 67, 68, 63, -1, 69, 60, 61, -1, 67, 63, 62, -1, 70, 64, 66, -1, 71, 70, 66, -1, 72, 69, 61, -1, 73, 69, 72, -1, 74, 68, 67, -1, 74, 75, 68, -1, 76, 70, 71, -1, 77, 73, 72, -1, 78, 76, 71, -1, 79, 80, 54, -1, 81, 73, 77, -1, 79, 54, 65, -1, 82, 75, 74, -1, 82, 83, 75, -1, 84, 76, 78, -1, 85, 81, 77, -1, 85, 86, 81, -1, 87, 84, 78, -1, 88, 89, 83, -1, 88, 83, 82, -1, 90, 86, 85, -1, 90, 91, 86, -1, 92, 93, 80, -1, 94, 84, 87, -1, 92, 80, 79, -1, 95, 94, 87, -1, 96, 94, 95, -1, 97, 89, 88, -1, 98, 96, 95, -1, 97, 99, 89, -1, 100, 91, 90, -1, 101, 96, 98, -1, 102, 103, 93, -1, 102, 93, 92, -1, 104, 100, 90, -1, 105, 99, 97, -1, 106, 101, 98, -1, 105, 107, 99, -1, 108, 101, 106, -1, 109, 100, 104, -1, 110, 108, 106, -1, 111, 107, 105, -1, 112, 109, 104, -1, 111, 113, 107, -1, 114, 115, 103, -1, 116, 108, 110, -1, 114, 103, 102, -1, 117, 116, 110, -1, 118, 109, 112, -1, 119, 113, 111, -1, 119, 120, 113, -1, 121, 116, 117, -1, 122, 118, 112, -1, 123, 121, 117, -1, 124, 115, 114, -1, 124, 125, 115, -1, 126, 120, 119, -1, 127, 121, 123, -1, 128, 127, 123, -1, 126, 129, 120, -1, 130, 131, 125, -1, 130, 125, 124, -1, 132, 133, 131, -1, 132, 131, 130, -1, 134, 129, 126, -1, 135, 127, 128, -1, 136, 135, 128, -1, 137, 138, 118, -1, 137, 118, 122, -1, 139, 129, 134, -1, 140, 133, 132, -1, 141, 140, 132, -1, 142, 140, 141, -1, 143, 142, 141, -1, 144, 139, 134, -1, 145, 135, 136, -1, 146, 145, 136, -1, 147, 139, 144, -1, 148, 142, 143, -1, 149, 148, 143, -1, 150, 148, 149, -1, 151, 145, 146, -1, 152, 151, 146, -1, 153, 147, 144, -1, 153, 154, 147, -1, 155, 151, 152, -1, 156, 150, 149, -1, 157, 154, 153, -1, 158, 138, 137, -1, 157, 159, 154, -1, 160, 155, 152, -1, 161, 159, 157, -1, 161, 162, 159, -1, 163, 155, 160, -1, 164, 158, 137, -1, 165, 166, 162, -1, 165, 162, 161, -1, 167, 163, 160, -1, 168, 158, 164, -1, 169, 166, 165, -1, 169, 170, 166, -1, 171, 170, 169, -1, 171, 172, 170, -1, 173, 163, 167, -1, 174, 173, 167, -1, 175, 168, 164, -1, 176, 172, 171, -1, 176, 177, 172, -1, 178, 177, 176, -1, 178, 179, 177, -1, 180, 173, 174, -1, 181, 168, 175, -1, 182, 181, 175, -1, 183, 180, 174, -1, 184, 185, 179, -1, 184, 179, 178, -1, 186, 180, 183, -1, 187, 188, 185, -1, 187, 185, 184, -1, 189, 181, 182, -1, 190, 191, 188, -1, 190, 188, 187, -1, 192, 186, 183, -1, 193, 191, 190, -1, 194, 191, 193, -1, 195, 186, 192, -1, 196, 189, 182, -1, 197, 198, 194, -1, 199, 195, 192, -1, 197, 194, 193, -1, 200, 195, 199, -1, 201, 202, 198, -1, 201, 198, 197, -1, 203, 189, 196, -1, 204, 205, 202, -1, 204, 202, 201, -1, 206, 203, 196, -1, 207, 208, 205, -1, 207, 205, 204, -1, 209, 210, 208, -1, 211, 200, 199, -1, 211, 212, 200, -1, 209, 208, 207, -1, 213, 203, 206, -1, 214, 212, 211, -1, 214, 215, 212, -1, 216, 213, 206, -1, 217, 215, 214, -1, 217, 218, 215, -1, 219, 218, 217, -1, 219, 220, 218, -1, 17, 213, 216, -1, 18, 17, 216, -1, 9, 220, 219, -1, 9, 8, 220, -1
        ]
      }
    }
  ]
  name "track2.obj"
}
DEF Linebot E-puck {
  translation -0.37 0 1.28
  controller "linebot_controller"
  groundSensorsSlot [
    DistanceSensor {
      translation 0 -0.01 0
      name "gs0"
      type "infra-red"
    }
    DistanceSensor {
      name "gs1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.01 0
      name "gs2"
      type "infra-red"
    }
  ]
}
DEF Arena Solid {
  children [
    Solid {
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -1.5 0 -1.5
                1.5 0 -1.5
                -1.5 0 1.5
                1.5 0 1.5
              ]
            }
            texCoord TextureCoordinate {
              point [
                0 0
                6 0
                0 6
                6 6
              ]
            }
            coordIndex [
              0, 2, 3, 1, -1
            ]
            texCoordIndex [
              2, 0, 1, 3, -1
            ]
          }
        }
      ]
      name "floor"
      model "floor"
      boundingObject Plane {
        size 3 3
      }
      locked TRUE
    }
    Solid {
      translation 1.505 0.05 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            metalness 0.8
            normalMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -0.005 0.05 1.5
                0.005 0.05 1.5
                0.005 0.05 -1.5
                -0.005 0.05 -1.5
                -0.005 -0.05 -1.5
                0.005 -0.05 -1.5
                0.005 -0.05 1.5
                -0.005 -0.05 1.5
                -0.005 -0.05 1.5
                0.005 -0.05 1.5
                0.005 0.05 1.5
                -0.005 0.05 1.5
                0.005 -0.05 -1.5
                -0.005 -0.05 -1.5
                -0.005 0.05 -1.5
                0.005 0.05 -1.5
                -0.005 -0.05 -1.5
                -0.005 -0.05 1.5
                -0.005 0.05 1.5
                -0.005 0.05 -1.5
                0.005 -0.05 1.5
                0.005 -0.05 -1.5
                0.005 0.05 -1.5
                0.005 0.05 1.5
              ]
            }
            texCoord TextureCoordinate {
              point [
                0 0
                0.0033333333333333 0
                0.0033333333333333 1
                0 1
                0 0
                0.0033333333333333 0
                0.0033333333333333 1
                0 1
                0 0
                0.0033333333333333 0
                0.0033333333333333 0.033333333333333
                0 0.033333333333333
                0 0
                0.0033333333333333 0
                0.0033333333333333 0.033333333333333
                0 0.033333333333333
                0 0
                1 0
                1 0.033333333333333
                0 0.033333333333333
                0 0
                1 0
                1 0.033333333333333
                0 0.033333333333333
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall"
      model "box"
      boundingObject Box {
        size 0.01 0.1 3
      }
    }
    Solid {
      translation -1.505 0.05 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -0.005 0.05 1.5
                0.005 0.05 1.5
                0.005 0.05 -1.5
                -0.005 0.05 -1.5
                -0.005 -0.05 -1.5
                0.005 -0.05 -1.5
                0.005 -0.05 1.5
                -0.005 -0.05 1.5
                -0.005 -0.05 1.5
                0.005 -0.05 1.5
                0.005 0.05 1.5
                -0.005 0.05 1.5
                0.005 -0.05 -1.5
                -0.005 -0.05 -1.5
                -0.005 0.05 -1.5
                0.005 0.05 -1.5
                -0.005 -0.05 -1.5
                -0.005 -0.05 1.5
                -0.005 0.05 1.5
                -0.005 0.05 -1.5
                0.005 -0.05 1.5
                0.005 -0.05 -1.5
                0.005 0.05 -1.5
                0.005 0.05 1.5
              ]
            }
            texCoord TextureCoordinate {
              point [
                0 0
                0.0033333333333333 0
                0.0033333333333333 1
                0 1
                0 0
                0.0033333333333333 0
                0.0033333333333333 1
                0 1
                0 0
                0.0033333333333333 0
                0.0033333333333333 0.033333333333333
                0 0.033333333333333
                0 0
                0.0033333333333333 0
                0.0033333333333333 0.033333333333333
                0 0.033333333333333
                0 0
                1 0
                1 0.033333333333333
                0 0.033333333333333
                0 0
                1 0
                1 0.033333333333333
                0 0.033333333333333
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#2"
      model "box"
      boundingObject Box {
        size 0.01 0.1 3
      }
    }
    Solid {
      translation 0 0.05 1.505
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -1.51 0.05 0.005
                1.51 0.05 0.005
                1.51 0.05 -0.005
                -1.51 0.05 -0.005
                -1.51 -0.05 -0.005
                1.51 -0.05 -0.005
                1.51 -0.05 0.005
                -1.51 -0.05 0.005
                -1.51 -0.05 0.005
                1.51 -0.05 0.005
                1.51 0.05 0.005
                -1.51 0.05 0.005
                1.51 -0.05 -0.005
                -1.51 -0.05 -0.005
                -1.51 0.05 -0.005
                1.51 0.05 -0.005
                -1.51 -0.05 -0.005
                -1.51 -0.05 0.005
                -1.51 0.05 0.005
                -1.51 0.05 -0.005
                1.51 -0.05 0.005
                1.51 -0.05 -0.005
                1.51 0.05 -0.005
                1.51 0.05 0.005
              ]
            }
            texCoord TextureCoordinate {
              point [
                0 0
                1 0
                1 0.0033112582781457
                0 0.0033112582781457
                0 0
                1 0
                1 0.0033112582781457
                0 0.0033112582781457
                0 0
                1 0
                1 0.033112582781457
                0 0.033112582781457
                0 0
                1 0
                1 0.033112582781457
                0 0.033112582781457
                0 0
                0.0033112582781457 0
                0.0033112582781457 0.033112582781457
                0 0.033112582781457
                0 0
                0.0033112582781457 0
                0.0033112582781457 0.033112582781457
                0 0.033112582781457
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#3"
      model "box"
      boundingObject Box {
        size 3.02 0.1 0.01
      }
    }
    Solid {
      translation 0 0.05 -1.505
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -1.51 0.05 0.005
                1.51 0.05 0.005
                1.51 0.05 -0.005
                -1.51 0.05 -0.005
                -1.51 -0.05 -0.005
                1.51 -0.05 -0.005
                1.51 -0.05 0.005
                -1.51 -0.05 0.005
                -1.51 -0.05 0.005
                1.51 -0.05 0.005
                1.51 0.05 0.005
                -1.51 0.05 0.005
                1.51 -0.05 -0.005
                -1.51 -0.05 -0.005
                -1.51 0.05 -0.005
                1.51 0.05 -0.005
                -1.51 -0.05 -0.005
                -1.51 -0.05 0.005
                -1.51 0.05 0.005
                -1.51 0.05 -0.005
                1.51 -0.05 0.005
                1.51 -0.05 -0.005
                1.51 0.05 -0.005
                1.51 0.05 0.005
              ]
            }
            texCoord TextureCoordinate {
              point [
                0 0
                1 0
                1 0.0033112582781457
                0 0.0033112582781457
                0 0
                1 0
                1 0.0033112582781457
                0 0.0033112582781457
                0 0
                1 0
                1 0.033112582781457
                0 0.033112582781457
                0 0
                1 0
                1 0.033112582781457
                0 0.033112582781457
                0 0
                0.0033112582781457 0
                0.0033112582781457 0.033112582781457
                0 0.033112582781457
                0 0
                0.0033112582781457 0
                0.0033112582781457 0.033112582781457
                0 0.033112582781457
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#4"
      model "box"
      boundingObject Box {
        size 3.02 0.1 0.01
      }
    }
  ]
  name "rectangle arena"
  model "rectangle arena"
}
DEF can Solid {
  translation -0.87 0.06 0.96
  children [
    DEF canShape Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Cylinder {
        height 0.115
        radius 0.0325
      }
    }
  ]
  boundingObject USE canShape
  physics Physics {
    mass 0.013
  }
}
DEF can Solid {
  translation -0.87 0.06 0.83
  children [
    DEF canShape Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Cylinder {
        height 0.115
        radius 0.0325
      }
    }
  ]
  name "solid(1)"
  boundingObject USE canShape
  physics Physics {
    mass 0.013
  }
}
DEF can Solid {
  translation -0.87 0.06 0.69
  children [
    DEF canShape Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Cylinder {
        height 0.115
        radius 0.0325
      }
    }
  ]
  name "solid(2)"
  boundingObject USE canShape
  physics Physics {
    mass 0.013
  }
}
DEF can Solid {
  translation -0.87 0.06 0.55
  children [
    DEF canShape Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Cylinder {
        height 0.115
        radius 0.0325
      }
    }
  ]
  name "solid(3)"
  boundingObject USE canShape
  physics Physics {
    mass 0.013
  }
}
